google-apis-common/src/lib.rs

```diff
@@ -414,22 +413,26 @@ impl Read for MultiPartReader<'_> {
             }
             (0, true, true) => return Ok(0),
             (n, true, _) if n > 0 => {
+                use std::fmt::Write as _;
                 let (headers, reader) = self.raw_parts.remove(0);
+
+                let mut encoded_headers = String::new();
+                for (k, v) in &headers {
+                    if !encoded_headers.is_empty() {
+                        encoded_headers.push_str(LINE_ENDING);
+                    }
+
+                    write!(encoded_headers, "{}: {}", k, v.to_str().unwrap())
+                        .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?;
```

Comment:
Since Rust 1.74 this can be simplified to `io::Error::other`
https://doc.rust-lang.org/nightly/std/io/struct.Error.html#method.other

```suggestion
                        .map_err(|err| std::io::Error::other(err))?;
```

Reply:
A good call, thank you!